go/ics23.go

```diff
@@ -1,6 +1,6 @@
 /**
 This implements the client side functions as specified in
-https://github.com/cosmos/ics/tree/master/spec/ics-023-vector-commitments
+https://github.com/cosmos/ibc/tree/master/spec/core/ics-023-vector-commitments
```

Comment:
```suggestion
https://github.com/cosmos/ibc/tree/main/spec/core/ics-023-vector-commitments
```
Might as well update the branch name too while we're at it